packages/core/src/services/commits/promptCache.ts

```diff
@@ -71,10 +85,29 @@ export async function setCachedResponse({
   workspace: Workspace
   config: Config
   conversation: Conversation
-  response: ChainStepResponse<StreamType>
+  response: ChainStepResponse<T>
 }) {
   if (!shouldCache(config)) return
 
   const key = generateCacheKey(workspace, config, conversation)
-  await setToCache(key, response)
+
+  if (response.streamType === 'text') {
+    await setToCache<'text'>(key, {
+      streamType: 'text',
+      text: response.text,
+      usage: response.usage,
+      toolCalls: response.toolCalls,
+    })
+  } else if (response.streamType === 'object') {
+    await setToCache<'object'>(key, {
+      streamType: 'object',
+      text: response.text,
+      usage: response.usage,
+      object: response.object,
+    })
+  } else {
+    throw new Error(
+      'Invalid "streamType" response, it should be "text" or "object"',
+    )
+  }
```

Comment:
hmm i'm a bit worried we might miss something instead of picking the parameters you wanna cache why not cache everything except documentLogUuid and providerLog? This way we ensure there's no incompatibility between the chain response that is cached and the chain response that isn't cached

Reply:
What can we miss? Typescript is not complaining according to response types

Reply:
This throw is here in case we start producing a different type of response than `text` or `object` and we forget to handle it in the cache